### PR TITLE
OT Actor cleanup

### PR DESCRIPTION
**Pull request by sinui0**
This PR cleans up our OT actors a bit.

# Changes
- Docs, lints, clippy :paperclip: 
- Added `SpawnCompatExt` extension trait in `utils-aio` to support executor-agnostic spawning. This lets us simplify setup for the actors by passing a spawner in to the constructor. We may want to separate this out and release it as a standalone crate, I didn't see any that implements the same pattern (which was inspired by `async_compat`)
- Removed generic from `Sender/ReceiverControl`, it was unnecessary as they just implement traits anyhow.
- Removed generics from the actors in favor of `Box<dyn Trait>`
- Removed dependency on `actor-mux` and just uses our mock factory in `utils-aio`
- Provided helper methods for constructing the actors.

### CODE REVIEW

File: actors/actor-ot/src/receiver.rs [hunk 1]
```diff
@@ -39,58 +46,63 @@ pub struct KOSReceiverActor<T, U> {
...
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - The configuration for the receiver
+    /// * `addr` - The address of the receiver
+    /// * `spawner` - The spawner to spawn the internal tasks
+    /// * `channel` - The channel over which OT splits are synchronized with the remote
+    ///               KOSSenderActor
+    /// * `mux_control` - The muxer which sets up channels with the remote KOSSenderActor
     pub fn new(
```
Comment (th4s) [inline actors/actor-ot/src/receiver.rs:60]: Should we restrict this to `pub(crate)`, since we have the `create_` functions in `setup.rs`?

**Review by th4s — APPROVED** [review]
:rocket: nice, ack

File: utils/utils-aio/src/executor.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }
+}
+
+#[cfg(feature = "tokio_compat")]
+impl Spawn for Compat<tokio::runtime::Runtime> {
+    fn spawn_obj(
+        &self,
+        future: futures::future::FutureObj<'static, ()>,
+    ) -> Result<(), futures::task::SpawnError> {
+        drop(self.0.spawn(future));
```
[the remark this answers is not in the document]
Comment (th4s) [reply utils/utils-aio/src/executor.rs:49]: ahh ok I just removed the `drop` without adding the `let`. Anyways, let's keep it, was just curious

File: actors/actor-ot/src/config.rs [hunk 3]
```diff
@@ -1,33 +1,39 @@
...
-const DEFAULT_OT_ID: &str = "ParentOT";
-
+/// OT sender actor configuration
 #[derive(Debug, Default, Builder)]
 pub struct OTActorSenderConfig {
-    #[builder(default = "DEFAULT_OT_ID.to_string()")]
-    pub(crate) ot_id: String,
+    /// The ID of the sender
+    pub(crate) id: String,
+    /// The number of OTs to setup
```
Comment (themighty1) [inline actors/actor-ot/src/config.rs:8]: to set up :)

**Review by themighty1 — APPROVED** [review]
ack

**Pull request merged**